src/blocks/remote-data-container/edit.tsx

```diff
@@ -32,120 +32,97 @@ export function Edit( props: BlockEditProps< RemoteDataBlockAttributes > ) {
 	const {
 		getInnerBlocks,
 		getSupportedPatterns,
+		innerBlocksPattern,
 		insertPatternBlocks,
-		markReadyForInsertion,
-		resetReadyForInsertion,
-		showPatternSelection,
+		resetInnerBlocks,
 	} = usePatterns( blockName, rootClientId );
-	const { execute } = useRemoteData( blockName, DISPLAY_QUERY_KEY );
-	const [ initialLoad, setInitialLoad ] = useState< boolean >( true );
-
-	function fetchRemoteData( input: RemoteDataQueryInput, insertBlocks = true ) {
-		execute( input, true )
-			.then( remoteData => {
-				if ( remoteData ) {
-					updateRemoteData(
-						{
-							enabledOverrides: props.attributes.remoteData?.enabledOverrides ?? [],
-							...remoteData,
-						},
-						insertBlocks
-					);
-				}
-			} )
-			.catch( () => {} )
-			.finally( () => {
-				setInitialLoad( false );
-			} );
+
+	const { data, fetch, loading, reset } = useRemoteData( {
+		blockName,
+		externallyManagedRemoteData: props.attributes.remoteData,
+		externallyManagedUpdateRemoteData: updateRemoteData,
+		queryKey: DISPLAY_QUERY_KEY,
+	} );
+
+	const [ showPatternSelection, setShowPatternSelection ] = useState< boolean >( false );
+
+	function refreshRemoteData(): void {
+		void fetch( props.attributes.remoteData?.queryInput ?? {} );
 	}
 
-	// Update the remote data in the block attributes, which is passed via context
-	// to children blocks. If this is the initial load of remote data, show the
-	// pattern selection modal so that we can insert the blocks from the pattern.
-	function updateRemoteData( remoteData: RemoteData, insertBlocks = false ) {
-		if ( hasRemoteDataChanged( props.attributes.remoteData, remoteData ) ) {
-			props.setAttributes( { remoteData } );
-		}
+	function resetPatternSelection(): void {
+		resetInnerBlocks();
+		setShowPatternSelection( false );
+	}
 
-		if ( insertBlocks ) {
-			markReadyForInsertion();
-		}
+	function resetRemoteData(): void {
+		reset();
+		resetPatternSelection();
 	}
 
-	const hasInputVariables = Boolean(
-		blockConfig.selectors.find( selector => selector.query_key === DISPLAY_QUERY_KEY )?.inputs
-			?.length
-	);
+	function onSelectPattern( pattern: BlockPattern ): void {
+		insertPatternBlocks( pattern );
+		setShowPatternSelection( false );
+	}
 
-	function refreshRemoteData() {
-		if ( ! props.attributes.remoteData?.queryInput ) {
-			if ( hasInputVariables ) {
+	function onSelectRemoteData( queryInput: RemoteDataQueryInput ): void {
+		void fetch( queryInput ).then( () => {
```

Comment:
Would it make sense for `queryInput` to be an array? For a future where we allow multiple selection

Reply:
`queryInput` is an object that could have an array inside of it. (It's `Record<string, unknown>` right now)

Reply:
> Would it make sense for queryInput to be an array? For a future where we allow multiple selection

We would need a construct around that dispatches multiple queries for each selected item. Will be fun!